deploy/osd-master-node-labeller/10-osd-master-node-labeller.CronJob.yaml

```diff
@@ -0,0 +1,81 @@
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: osd-master-node-labeller
+  namespace: openshift-config
+spec:
+  failedJobsHistoryLimit: 3
+  successfulJobsHistoryLimit: 3
+  concurrencyPolicy: Replace
+  schedule: "0 0 * * *"
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          affinity:
+            nodeAffinity:
+              preferredDuringSchedulingIgnoredDuringExecution:
+                - preference:
+                    matchExpressions:
+                      - key: node-role.kubernetes.io/infra
+                        operator: Exists
+                  weight: 1
+          tolerations:
+            - effect: NoSchedule
+              key: node-role.kubernetes.io/infra
+              operator: Exists
+          serviceAccountName: osd-master-node-labeller
+          restartPolicy: Never
+          containers:
+            - name: osd-master-node-labeller
+              image: image-registry.openshift-image-registry.svc:5000/openshift/cli:latest
+              imagePullPolicy: Always
+              command:
+                - /bin/bash
+                - -c
+                - |
+                  #!/bin/bash
+                  set -euo pipefail
+
+                  echo "Starting master node labeller job..."
+
+                  # Get all nodes with node-role.kubernetes.io/master label
+                  master_nodes=$(oc get nodes -l node-role.kubernetes.io/master -o jsonpath='{.items[*].metadata.name}')
+
+                  if [ -z "$master_nodes" ]; then
+                    echo "No nodes found with label node-role.kubernetes.io/master"
+                    exit 0
+                  fi
+
+                  echo "Found master nodes: $master_nodes"
+
+                  # Check if all master nodes already have control-plane label
+                  nodes_needing_label=0
+                  for node in $master_nodes; do
+                    if ! oc get node "$node" --show-labels | grep -q "node-role.kubernetes.io/control-plane"; then
```

Comment:
why the script loop the node twice to check labels (here in `line 55` and `line 72`)

Reply:
yes, we can count the node with label directly without looping